swap_meet/vendor.py

```diff
@@ -1,2 +1,233 @@
+        ----------
+        other_vendor : Vendor
+            The other participant in the swap.
+        own_item : Item
+        swap_item : Item
+            The items to be swapped. 'own_item' must be present in the local
+            'inventory' and 'swap_item' must be present in the 'inventory'
+            attribute of the 'other_vendor' object. 
+        '''
+
+        try:
+            own_index = self.inventory.index(own_item)
+            other_index = other_vendor.inventory.index(swap_item)
+        except ValueError:         # item not found in appropriate inventory
+            return False
+        
+        # handoff by index
+        self.inventory[own_index], other_vendor.inventory[other_index] = \
+                other_vendor.inventory[other_index], self.inventory[own_index]
+        return True
+    
+    def swap_first_item(self,other_vendor):
+        '''Swaps first item in inventory with another vendor's first item.
+        
+        Returns False if either inventory is empty. Otherwise, returns True.
+
+        Parameters
+        ----------
+        other_vendor : Vendor
+            The other participant in the swap.
+        '''
+
+        if not (self.inventory and other_vendor.inventory):
+            return False
+        
+        self.inventory[0], other_vendor.inventory[0] = \
+                                other_vendor.inventory[0], self.inventory[0]
+        return True
+    
+    def get_best_by_category(self, category):
+        '''Returns the item in the best condition of the input category.
+        
+        Parameters
+        ----------
+        category : str
+            The value by which to filter items in the inventory. 
+        '''
+
+        items = self.get_by_category(category)
+        if len(items) == 0:
+            return None
+        ratings = [item.condition for item in items]
+        return items[ratings.index(max(ratings))]
+    
+    def swap_best_by_category(self, other, my_priority, their_priority):
+        '''Swaps item with another vendor based on category and condition.
+        
+        Looks for the highest rated item in the other vendor's inventory
+        with a category matching 'my_priority' and the highest rated item
+        in the local inventory with a category matching 'their_priority'.
+        If found the two items are swapped between the two inventories.
+
+        Returns False if two appropriate items aren't found. Otherwise
+        returns True.
+
+        Parameters
+        ----------
+        other : Vendor
+            The other participant in the swap.
+        my_priority : str
+        their_priority : str
+            The two categories by which to filter items
+        '''
+        
+        own_item = self.get_best_by_category(their_priority)
+        swap_item = other.get_best_by_category(my_priority)
+        
+        if not (own_item and swap_item):
+            return False
+        return self.swap_items(other,own_item,swap_item)
```

Comment:
Because of how `swap_items` is implemented, checking if own_item and swap_item are valid here isn't necessary. You do that checking in swap_items(). `swap_items()` returns True if swapping happened and False if no swapping happened so you can remove 231 and 232.

```python
def swap_best_by_category(self, other, my_priority, their_priority):
        own_item = self.get_best_by_category(their_priority)
        swap_item = other.get_best_by_category(my_priority)
        return self.swap_items(other, my_best, their_best)
```